kotest-assertions/src/jvmMain/kotlin/io/kotest/matchers/CompilationMatchers.kt

```diff
@@ -0,0 +1,66 @@
+package io.kotest.matchers
+
+import com.tschuchort.compiletesting.KotlinCompilation
+import com.tschuchort.compiletesting.KotlinCompilation.ExitCode
+import com.tschuchort.compiletesting.KotlinCompilation.Result
+import com.tschuchort.compiletesting.SourceFile
+import io.kotest.Matcher
+import io.kotest.MatcherResult
+import io.kotest.should
+import io.kotest.shouldNot
+import java.io.File
+
+private fun compileCodeSnippet(codeSnippet: String): Result {
+   val kotlinCompilation = KotlinCompilation()
+      .apply {
+         sources = listOf(SourceFile.kotlin("KClass.kt", codeSnippet))
+         inheritClassPath = true
+      }
+   val compilationResult = kotlinCompilation.compile()
+   kotlinCompilation.workingDir.deleteRecursively()
+
+   return compilationResult
+}
+
+private val compiles = object : Matcher<String> {
+   override fun test(value: String): MatcherResult {
+      val compilationResult = compileCodeSnippet(value)
+      return MatcherResult(
+         compilationResult.exitCode == ExitCode.OK,
+         { "Expected code to compile, but it failed to compile" },
```

Comment:
I would include the actual underlying error as well.

Reply:
Made compilation non verbose, redirected compilation log to a ByteArrayOutputStream instead of System.Out and uses Error message of Result in matcher.